vic/drivers/shared_image/src/vic_store.c

```diff
@@ -48,25 +48,44 @@ vic_store(dmy_struct *dmy_current,
     size_t                     k;
     size_t                     m;
     size_t                     p;
+    double                     offset;
+    double                     time_num;
+    double                     end_time_num;
     int                       *ivar = NULL;
     double                    *dvar = NULL;
     size_t                     d2start[2];
     size_t                     d3start[3];
     size_t                     d4start[4];
     size_t                     d5start[5];
     size_t                     d6start[6];
+    dmy_struct                 end_time_date;
     nc_file_struct             nc_state_file;
     nc_var_struct             *nc_var;
 
     set_nc_state_file_info(&nc_state_file);
 
+    // advance dmy_current by one timestep since dmy_current is the 
+    // timestep-beginning timestamp, and state file date should be 
+    // the end of the current time step 
+    dt_seconds_to_time_units(global_param.time_units, global_param.dt, 
+                             &offset);
+    time_num = date2num(global_param.time_origin_num, dmy_current, 0, 
+                        global_param.calendar, global_param.time_units);
+    end_time_num = time_num + offset; 
+
+    // allocate dmy struct for end of current time step 
+    num2date(global_param.time_origin_num, end_time_num, 0., 
+                             global_param.calendar, global_param.time_units, 
+                             &end_time_date);
+    
+
```

Comment:
Actually, on second thought, the timestamp will be incorrect if it's set at initialization and not at the time at which we write the state file. But I still think it seems like this shouldn't be in `vic_store`, since that's a shared_image function.